c7n/filters/core.py

```diff
@@ -493,6 +493,11 @@ def process_value_type(self, sentinel, value, resource):
         elif self.vtype == 'age':
             if not isinstance(sentinel, datetime.datetime):
                 sentinel = datetime.datetime.now(tz=tzutc()) - timedelta(sentinel)
+            if isinstance(value, str):
```

Comment:
rather than cast strings into ints, I think it would be better to change line 503 to cast `value` to a float when converting to a datetime obj as the timestamp is a floating point number of seconds from the epoch, i.e.

```python
if isinstance(value, (str, int, float)):
  try:
    value = datetime.datetime.fromtimestamp(float(value)).replace(tzinfo=tzutc())
```

Reply:
sounds good to me.